src/choices.c

```diff
@@ -39,31 +42,33 @@ min_match_length(char *str, char *query)
 }
 
 float
-score_str(char *str, char *query)
+score_str(char *str, char *query, int *mpos, size_t *mlen)
 {
 	size_t slen;
 	size_t qlen;
-	size_t mlen;
 
 	slen = strlen(str);
 	qlen = strlen(query);
 	if (qlen == 0)
-		return 1;	
+		return 1;
 	if (slen == 0)
 		return 0;
-	if ((mlen = min_match_length(str, query)) == 0)
+	if ((*mlen = min_match_length(str, query, mpos)) == 0)
 		return 0;
-	return (float)qlen / (float)mlen / (float)slen;
+	return (float)qlen / (float)*mlen / (float)slen;
 }
 
-
 void
 choices_score(struct choices *cs, char *query)
 {
 	struct choice *c;
+	int mpos = 0;
+	size_t mlen = 0;
```

Comment:
I see how `mpos` is reset on every iteration in `score_str` but I don't get if `mlen` will reset as well? It seems like it will only reset if it is `-1`?